x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -64,6 +72,50 @@ public static Query toQuery(Expression e, TranslatorHandler handler) {
         throw new QlIllegalArgumentException("Don't know how to translate {} {}", e.nodeName(), e);
     }
 
+    public static class InsensitiveBinaryComparisons extends ExpressionTranslator<InsensitiveBinaryComparison> {
+
+        @Override
+        protected Query asQuery(InsensitiveBinaryComparison bc, TranslatorHandler handler) {
+            return doTranslate(bc, handler);
+        }
+
+        public static void checkInsensitiveComparison(InsensitiveBinaryComparison bc) {
+            Check.isTrue(bc.right().foldable(),
+                "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [{}]",
+                bc.right().sourceLocation().getLineNumber(), bc.right().sourceLocation().getColumnNumber(),
+                Expressions.name(bc.right()), bc.symbol());
+        }
+
+        public static Query doTranslate(InsensitiveBinaryComparison bc, TranslatorHandler handler) {
+            checkInsensitiveComparison(bc);
+            return handler.wrapFunctionQuery(bc, bc.left(), translate(bc, handler));
+        }
+
+        private static Query translate(InsensitiveBinaryComparison bc, TranslatorHandler handler) {
+            Source source = bc.source();
+            String name = handler.nameOf(bc.left());
+            Object value = valueOf(bc.right());
+
+            if (bc instanceof InsensitiveEquals || bc instanceof InsensitiveNotEquals) {
+                if (bc.left() instanceof FieldAttribute) {
+                    // equality should always be against an exact match
+                    // (which is important for strings)
+                    name = ((FieldAttribute) bc.left()).exactAttribute().name();
+                }
+                Query query;
```

Comment:
`Query query = new TermQuery.....` (meaning, these two lines should be placed on a single one instead.